app/controllers/tool_groups_controller.rb

```diff
@@ -0,0 +1,91 @@
+class ToolGroupsController < ApplicationController
+  before_action :authorize!
+  before_action :convert_hyphen_to_dash, only: [:create, :update]
+
+  def index
+    render json: tool_groups_ordered_by_name, include: params[:include], fields: field_params, status: :ok
+  end
+
+  def create
+    create_tool_group
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def create_tool
+    tool_group_id = params[:tool_group_id].to_i
+    resource_id = params.dig(:data, :relationships, :tool, :data, :id).to_i
+    suggestions_weight = params.dig(:data, :attributes, "suggestions-weight").to_f
+
+    resource_tool_group = ResourceToolGroup.create!(
+      tool_group_id: tool_group_id,
+      resource_id: resource_id,
+      suggestions_weight: suggestions_weight
+    )
+
+    response.headers["Location"] = "tool-groups/#{resource_tool_group.id}"
+    render json: resource_tool_group, status: :created
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def update_tool
+    existing = resource_tool_group
+    existing.update!(
+      resource_id: params[:data][:attributes]["resource-id"],
```

Comment:
the resource id should be read from the relationships json.

```
curl --location --request PUT 'https://mobile-content-api-stage.cru.org/tool-groups/1/tools/5' \
--header 'Content-Type: application/vnd.api+json' \
--header 'Accept: application/vnd.api+json' \
--header 'Authorization: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxLCJleHAiOjE2OTExODE4MDUsImlhdCI6MTY5MTA5NTQwNX0.UVBAMJg0Gq8eX3taz9I1REJevJER05y4DNo-mInhaVM' \
--data '{
 "data":   {
   "type": "tool-group-tool",
   "attributes": {
    "suggestions-weight": "1.0"
   },
   "relationships": {
    "tool": {
     "data": {
      "type": "resource",
      "id": 1
     }
    }
   }
 }
}'
```

Reply:
this should be solved now too